packages/node_modules/@webex/internal-plugin-ediscovery/src/report-generator.js

```diff
@@ -46,29 +59,32 @@ const ReportGenerator = SparkPlugin.extend({
       throw Error('Invalid activity');
     }
 
+    const date = new Date(activity.timestamp);
     const mime = new MimeBuilder('multipart/mixed')
       .setHeader('From', activity.actorDisplayName)
-      .setHeader('Subject', activity.activityId)
-      .setHeader('Date', activity.timestamp);
+      .setHeader('To', activity.participantDisplayNames)
+      .setHeader('Subject', activity.spaceName)
+      .setHeader('Date', date.toUTCString());
 
     let content = activity.objectDisplayName;
     if (activity.verb !== 'post') {
       content = activity.verb;
       // Include the target of the event. e.g. Add Person
-      if (activity.objectId) {
-        content += ` ${activity.objectId}`;
+      if (activity.objectDisplayName) {
+        content += ` ${activity.objectDisplayName}`;
       }
     }
 
     mime.createChild('text/plain')
       .setContent(content);
 
+    // Attach file and whiteboard shares
     if (activity.verb === 'share' && activity.files) {
-      for (let i = 0; i < activity.files.length; i += 1) {
-        const file = activity.files[i];
-        if (file.name && file.content) {
-          mime.createChild(file.mimeType, {filename: file.name})
-            .setContent(file.content)
+      const files = activity.files;
+      for (let i = 0; i < files.length; i += 1) {
```

Comment:
If you don't need `i` for reporting or logging, you could use a map or other looping mechanism here so you don't have to keep track of it.